src/codehelp/signaturehelpengine.vala

```diff
@@ -148,21 +147,139 @@ namespace Vls.SignatureHelpEngine {
 
             if (data_type is Vala.MethodType) {
                 var mt = (Vala.MethodType) data_type;
+                var context = Vala.CodeContext.get ();
```

Comment:
Perhaps instead of this, we could add a `Compilation compilation` argument to `show_help()` (like what we already have with `show_help_with_updated_context()`, and then you can get the context with `var context = compilation.code_context`. I just like my functions to be as pure as possible. It makes the code easier to follow.